### PR TITLE
Fix background folder fetch pushing requests to wrong queue

### PR DESCRIPTION
**Pull request by Ansariel**
Actual fix is lines 388 and 395

### CODE REVIEW

**Review by akleshchev — APPROVED** [review]

**Review by marchcat — APPROVED** [review]
Thank you!

File: indra/newview/llinventorymodelbackgroundfetch.cpp [hunk 1]
```diff
@@ -254,35 +254,35 @@ bool LLInventoryModelBackgroundFetch::folderFetchActive() const
...
 }
 
-void LLInventoryModelBackgroundFetch::addRequestAtFront(const LLUUID & id, bool recursive, bool is_category)
+void LLInventoryModelBackgroundFetch::addRequestAtFront(const LLUUID& id, bool recursive, bool is_category)
 {
     EFetchType recursion_type = recursive ? FT_RECURSIVE : FT_DEFAULT;
     if (is_category)
     {
-        mFetchFolderQueue.push_front(FetchQueueInfo(id, recursion_type, is_category));
+        mFetchFolderQueue.emplace_front(FetchQueueInfo(id, recursion_type, is_category));
```
Comment (brad-linden) [inline indra/newview/llinventorymodelbackgroundfetch.cpp:262]: shouldn't all these `emplace()` variants usages avoid calling the constructor and just directly take the constructor's args?

**Review by brad-linden — COMMENTED** [review]
I'm thinking that we will likely want to merge this into `develop` rather than DeltaFPS for now

File: indra/newview/llinventorymodelbackgroundfetch.cpp (hunk 1 above)
Comment (Ansariel) [reply indra/newview/llinventorymodelbackgroundfetch.cpp]: That definitely makes sense, yes. I updated the PR accordingly.

**Comment by Ansariel** [comment]
> I'm thinking that we will likely want to merge this into `develop` rather than DeltaFPS for now

Isn't DeltaFPS targetted to become next release? That's the reason I created the PR against this branch, especially since this bug is currently out in the wild. If you prefer it going into develop, I can retarget the PR though.

**Comment by brad-linden** [comment]
> Isn't DeltaFPS targetted to become next release? That's the reason I created the PR against this branch, especially since this bug is currently out in the wild. If you prefer it going into develop, I can retarget the PR though.

Yes, but it is frozen for stabilization.  In general with the new git flow process we won't be committing things directly to the `release/foo` branches unless the issue being fixed is specifically flagged as a showstopper by QA or product review.  We are hoping to get DeltaFPS quickly and have at least one more performance focused release to follow.

**Review by brad-linden — APPROVED** [review]

**Pull request merged**